### PR TITLE
Fix a few issues on assert/double error/legacy constructor

### LINKED ISSUE #77960: expected key in DispatchKeySet(CPU, CUDA, ....) but got: MPS
### 🐛 Describe the bug
Error:
```python
RuntimeError: new(): expected key in DispatchKeySet(CPU, CUDA, HIP, XLA, IPU, XPU, HPU, Lazy) but got: MPS
```
This error is thrown from hugging face transformers module.

https://github.com/huggingface/transformers/blob/54192058f3826eb38f9aaea02961f1304678198f/src/transformers/generation_utils.py#L1658-L1659

minimum reproducible would presumably be:
```python 
tensor  = torch.LongTensor(...)
tensor.new(1, device=torch.device('mps'))
```

### Versions

```
PyTorch version: 1.12.0.dev20220520
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.23.1
Libc version: N/A

Python version: 3.10.4 (main, Mar 31 2022, 03:37:37) [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.12.0.dev20220520
[conda] numpy                     1.22.3                   pypi_0    pypi
[conda] torch                     1.12.0.dev20220520          pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by albanD**
Fixes https://github.com/pytorch/pytorch/issues/77960, https://github.com/pytorch/pytorch/issues/77957, https://github.com/pytorch/pytorch/issues/77781

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -5076,6 +5076,11 @@ def callable(a, b) -> number
 are like normal tensors, but they carry no data.
 """)
 
+add_docstr_all('is_mps',
+               r"""
+Is ``True`` if the Tensor is stored MPS, ``False`` otherwise.
```
Comment (soulitzer) [inline torch/_tensor_docs.py:5081]: "stored on the MPS"?

**Review by soulitzer — APPROVED** [review]

File: aten/src/ATen/native/mps/OperationUtils.mm [hunk 2]
```diff
@@ -142,7 +145,7 @@ MPSDataType getMPSScalarType(ScalarType scalar_type) {
     case ScalarType::Bool:
       return MPSDataTypeBool;
     default:
-      TORCH_INTERNAL_ASSERT(false, "Trying to convert ", scalar_type, " to the MPS backend but there is no mapping for it.")
+      TORCH_INTERNAL_ASSERT(false, "Trying to convert ", scalar_type, " to the MPS backend but that backend doesn't exist.")
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/OperationUtils.mm:148]: I don't think this is the right phrasing. We can just say that `but MPS backend doesn't have support for it`

**Comment by kulinseth** [comment]
Modulo the nit on updating the comment the PR looks good to me.

**Review by kulinseth — APPROVED** [review]

File: aten/src/ATen/native/mps/OperationUtils.mm (hunk 2 above)
Comment (albanD) [reply aten/src/ATen/native/mps/OperationUtils.mm:148]: This is an internal assert as all the dtypes should be handled above.
The only way for this to trigger is if someone adds a new dtype but does not update this.
But I guess there are weird quantized dtypes that we don't support.
Let me change that back to a non-internal then and use your proposed phrasing.

Comment (albanD) [reply aten/src/ATen/native/mps/OperationUtils.mm:148]: This is an internal assert as all the dtypes should be handled above.
The only way for this to trigger is if someone adds a new dtype but does not update this.
But I guess there are weird quantized dtypes that we don't support.
Let me change that back to a non-internal then and use your proposed phrasing.

File: aten/src/ATen/native/mps/OperationUtils.mm [hunk 3]
```diff
@@ -117,8 +117,11 @@ MPSDataType getMPSDataType(ScalarType scalar_type) {
       return MPSDataTypeInt8;
     case ScalarType::Bool:
       return MPSDataTypeBool;
+    case ScalarType::Double:
+      TORCH_CHECK_TYPE(false, "Cannot convert a float64 Tensor to MPS as the MPS framework doesn't support float64. "
+                       "Please use float32 instead.")
     default:
-      TORCH_CHECK_TYPE(false, "Trying to convert ", scalar_type, " to the MPS backend but there is no mapping for it.")
+      TORCH_INTERNAL_ASSERT(false, "Trying to convert ", scalar_type, " to the MPS backend but that backend doesn't exist.")
```
[the remark this answers is not in the document]
Comment (albanD) [reply aten/src/ATen/native/mps/OperationUtils.mm:124]: I was expecting this to be exhaustive. But it is not, will fix.

**Pull request closed without merging**